tests/test_applets/batchProcessing/test_batchProcessingGui.py

```diff
@@ -0,0 +1,203 @@
+
+
+@pytest.fixture()
+def role_names():
+    return ["testRoleName1", "testRoleName2", "testRoleName3"]
+
+
+@pytest.fixture()
+def parent_applet(role_names):
+    parent_applet = mock.Mock()
+    parent_applet.dataSelectionApplet.topLevelOperator.DatasetRoles.value = role_names
+    parent_applet.run_export = mock.Mock()
+
+    return parent_applet
+
+
+@pytest.fixture()
+def batch_processing_gui(qtbot, parent_applet):
+    return construct_widget(qtbot, BatchProcessingGui, parent_applet)
+
+
+@pytest.fixture
+def filenames():
+    return ["file1", "file2"]
+
+
+@pytest.fixture
+def drag_n_drop_event(filenames):
+    dndevent = mock.Mock()
+    urls = [QUrl(x) for x in filenames]
+    for url in urls:
+        url.isLocalFile = mock.Mock(return_value=True)
+    dndevent.mimeData.return_value.has_urls.return_value = True
+    dndevent.mimeData.return_value.urls.return_value = urls
+    return dndevent
+
+
+def test_FileListWidget_drag_n_drop(file_list_widget, drag_n_drop_event, filenames):
+    assert file_list_widget.count() == 0
+
+    file_list_widget.dragEnterEvent(drag_n_drop_event)
+    assert drag_n_drop_event.acceptProposedAction.called_once()
+    assert file_list_widget.count() == 0
+
+    file_list_widget.dropEvent(drag_n_drop_event)
+    assert file_list_widget.count() == len(filenames)
+    for url in filenames:
+        items = file_list_widget.findItems(url, Qt.MatchFixedString)
+        assert len(items) == 1
+
+
+def test_BatchRoleWidget(qtbot, batch_role_widget, filenames):
+    assert batch_role_widget.select_button.text() == f"Select TestRoleName Files..."
+    assert batch_role_widget.clear_button.text() == f"Clear TestRoleName Files"
+
+    assert batch_role_widget.list_widget.count() == 0
+
+    batch_role_widget.list_widget.addItems(filenames)
+    assert batch_role_widget.filepaths == filenames
+
+    qtbot.mouseClick(batch_role_widget.clear_button, Qt.LeftButton, delay=1)
+    assert len(batch_role_widget.filepaths) == 0
+
+
+class RequestMock:
+    _instances = []
+
+    def __new__(cls, *args, **kwargs):
+        instance = mock.Mock()
+        cls._instances.append(instance)
+        return instance
+
+
+def assert_buttons_in_init_state(batch_processing_gui):
+    assert batch_processing_gui.run_button.isEnabled() is True
+    assert batch_processing_gui.cancel_button.isVisible() is False
+
+
+def assert_buttons_in_running_state(batch_processing_gui):
+    assert batch_processing_gui.run_button.isEnabled() is False
+    assert batch_processing_gui.cancel_button.isVisible() is True
+    assert batch_processing_gui.cancel_button.isEnabled() is True
+
+
+def test_BatchProcessingGui_execution(qtbot, batch_processing_gui, role_names, filenames):
+    assert batch_processing_gui.count() == len(role_names)
+    assert_buttons_in_init_state(batch_processing_gui)
+
+    with mock.patch("ilastik.applets.batchProcessing.batchProcessingGui.Request", new=RequestMock):
```

Comment:
This could cause potential problems because `_instances` is not cleared after the function exit.
We could use a locally created callable that accumulates instances:

```python
instances = []

def RequestMock():
    instances.append(mock.Mock())
    return instances[-1]

with ...:
    ...
    assert len(instances) == 0
```